ports/mimxrt/psram.c

```diff
@@ -0,0 +1,237 @@
+    flexspi2_command(0, addr);     // exit quad mode
+    flexspi2_command(1, addr);     // reset enable
+    flexspi2_command(2, addr);     // reset (is this really necessary?)
+    uint32_t id = flexspi2_psram_id(addr);
+
+    switch (id & 0xFFFF)
+    {
+        default:
+            break;
+
+        case 0x5D0D:         // AP / Ipus / ESP / Lyontek
+            result = 8;
+            break;
+
+        case 0x5D9D:         // ISSI
+            switch ((id >> 21) & 0x7)             // get size (Datasheet Table 6.2)
+            {
+                case 0b011:
+                    result = 8;
+                    break;
+                case 0b100:
+                    result = 16;
+                    break;
+            }
+            break;
+    }
+
+    return result;
+}
+
+size_t configure_external_ram() {
+    // initialize pins
+    IOMUXC->SW_PAD_CTL_PAD[22] = 0x1B0F9;     // 100K pullup, strong drive, max speed, hyst
+    IOMUXC->SW_PAD_CTL_PAD[23] = 0x110F9;     // keeper, strong drive, max speed, hyst
+    IOMUXC->SW_PAD_CTL_PAD[24] = 0x1B0F9;     // 100K pullup, strong drive, max speed, hyst
+    IOMUXC->SW_PAD_CTL_PAD[25] = 0x100F9;     // strong drive, max speed, hyst
+    IOMUXC->SW_PAD_CTL_PAD[26] = 0x170F9;     // 47K pullup, strong drive, max speed, hyst
+    IOMUXC->SW_PAD_CTL_PAD[27] = 0x170F9;     // 47K pullup, strong drive, max speed, hyst
+    IOMUXC->SW_PAD_CTL_PAD[28] = 0x170F9;     // 47K pullup, strong drive, max speed, hyst
+    IOMUXC->SW_PAD_CTL_PAD[29] = 0x170F9;     // 47K pullup, strong drive, max speed, hyst
+
+    IOMUXC->SW_MUX_CTL_PAD[22] = 8 | 0x10;     // ALT1 = FLEXSPI2_A_SS1_B (Flash)
+    IOMUXC->SW_MUX_CTL_PAD[23] = 8 | 0x10;     // ALT1 = FLEXSPI2_A_DQS
+    IOMUXC->SW_MUX_CTL_PAD[24] = 8 | 0x10;     // ALT1 = FLEXSPI2_A_SS0_B (RAM)
+    IOMUXC->SW_MUX_CTL_PAD[25] = 8 | 0x10;     // ALT1 = FLEXSPI2_A_SCLK
+    IOMUXC->SW_MUX_CTL_PAD[26] = 8 | 0x10;     // ALT1 = FLEXSPI2_A_DATA0
+    IOMUXC->SW_MUX_CTL_PAD[27] = 8 | 0x10;     // ALT1 = FLEXSPI2_A_DATA1
+    IOMUXC->SW_MUX_CTL_PAD[28] = 8 | 0x10;     // ALT1 = FLEXSPI2_A_DATA2
+    IOMUXC->SW_MUX_CTL_PAD[29] = 8 | 0x10;     // ALT1 = FLEXSPI2_A_DATA3
+
+    IOMUXC->SELECT_INPUT_1[kIOMUXC_FLEXSPI2_IPP_IND_DQS_FA_SELECT_INPUT] = 1;     // GPIO_EMC_23 for Mode: ALT8, pg 986
+    IOMUXC->SELECT_INPUT_1[kIOMUXC_FLEXSPI2_IPP_IND_IO_FA_BIT0_SELECT_INPUT] = 1;     // GPIO_EMC_26 for Mode: ALT8
+    IOMUXC->SELECT_INPUT_1[kIOMUXC_FLEXSPI2_IPP_IND_IO_FA_BIT1_SELECT_INPUT] = 1;     // GPIO_EMC_27 for Mode: ALT8
+    IOMUXC->SELECT_INPUT_1[kIOMUXC_FLEXSPI2_IPP_IND_IO_FA_BIT2_SELECT_INPUT] = 1;     // GPIO_EMC_28 for Mode: ALT8
+    IOMUXC->SELECT_INPUT_1[kIOMUXC_FLEXSPI2_IPP_IND_IO_FA_BIT3_SELECT_INPUT] = 1;     // GPIO_EMC_29 for Mode: ALT8
+    IOMUXC->SELECT_INPUT_1[kIOMUXC_FLEXSPI2_IPP_IND_SCK_FA_SELECT_INPUT] = 1;     // GPIO_EMC_25 for Mode: ALT8
+
+    // turn on clock  (QSPI flash & PSRAM chips usually spec max clock 100 to 133 MHz)
```

Comment:
The clock setup in the following lines down to  line 137 inc. can be made a little bit more readable using the NXP lib macros:
```
    // Set FLEXSPI2_PODF divide by 5.
    CLOCK_SetDiv(kCLOCK_Flexspi2Div, 4);
    // Set Flexspi2 clock source to PLL2, 528 MHz.
    CLOCK_SetMux(kCLOCK_Flexspi2Mux, 3);
    CLOCK_EnableClock(kCLOCK_FlexSpi2);
```
The various other combinations may be added as a comment: